File: backup_extensions.py
import getpass
import os

username = getpass.getuser()
file_path= "/home/"+username+"/Documents/fresh_linux_install/extensions.txt"
extensions_path = "/home/"+username+"/.local/share/gnome-shell/extensions/"
entries = os.listdir(extensions_path)
with open(file_path,"w") as f:
    for entry in entries:
        f.write(entry+"\n")

File: backup_extensions.js
const fs = require("fs");

let dir = "/home/hussein/.local/share/gnome-shell/extensions"
let files = fs.readdirSync(dir);

var stream = fs.createWriteStream("/home/hussein/Documents/fresh_linux_install/extensions.txt");
stream.once('open', function(fd) {
  files.forEach(file=>{
    stream.write(`${file}\n`);
  }) 
  stream.end();
});

